OpenRA.Game/Activities/Activity.cs

```diff
@@ -200,8 +200,18 @@ public Activity TickOuter(Actor self)
 		/// </summary>
 		protected virtual void OnLastRun(Actor self) { }
 
+		/// <summary>
+		/// Runs once on Actor.Dispose() and can be used to perform activity clean-up on actor death/disposal,
+		/// for example by force-triggering OnLastRun (which would otherwise be skipped).
+		/// </summary>
+		public virtual void OnActorDispose(Actor self) { }
+
 		public virtual bool Cancel(Actor self, bool keepQueue = false)
 		{
+			// Just a safe-guard in case this activity was already successfully cancelled
+			if (IsCanceled)
```

Comment:
This will cause problems if you cancel an activity that takes a few ticks to complete (e.g. Move), queue a new activity, and then call Cancel again before the original activity has completed. Expected behaviour: second activity is cancelled, actual behaviour: it won't be.